recipe/build.sh

```diff
@@ -40,6 +40,10 @@ if [[ ${cuda_compiler_version} != "None" ]]; then
         export TORCH_CUDA_ARCH_LIST="3.5;5.0;6.0;6.1;7.0;7.5;8.0;8.6+PTX"
     elif [[ ${cuda_compiler_version} == 11.2 ]]; then
         export TORCH_CUDA_ARCH_LIST="3.5;5.0;6.0;6.1;7.0;7.5;8.0;8.6+PTX"
+    elif [[ ${cuda_compiler_version} == 11.8 ]]; then
+        export TORCH_CUDA_ARCH_LIST="3.5;5.0;6.0;6.1;7.0;7.5;8.0;8.6;8.9+PTX"
+    elif [[ ${cuda_compiler_version} == 12.0 ]]; then
+        export TORCH_CUDA_ARCH_LIST="5.0;6.0;6.1;7.0;7.5;8.0;8.6;8.9;9.0+PTX"
```

Comment:
Since we are adding CUDA 12 here, would it make sense to add the migrator from PR ( https://github.com/conda-forge/dgl-feedstock/pull/26 ) too?

Or would you rather do that separately?

Reply:
Yes, we'll tackle that separately. I accidentally added a couple migrations here by rerendering 😬

Reply:
The bot refreshed the CUDA 12 migrator in PR: https://github.com/conda-forge/dgl-feedstock/pull/31

Anything else we need can go there